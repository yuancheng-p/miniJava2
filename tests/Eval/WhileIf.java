public class WhileIf {
  public static void main (String [] args) {
    int sum = 1, i = 1;
    // calculate 10!
    while (i <= 10) {
      sum = sum * i++;
    }

    int result = 0;
    if (sum > 5050) {
      result = sum;
    } else {
      result = 5050;
    }

    int r = result;

  }
}
