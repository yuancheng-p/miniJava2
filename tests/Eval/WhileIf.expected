successfull parsing
start typing
check add and methods:WhileIf:main
--------- env --------
Boolean : class_env(parent=: Object)
Integer : class_env(parent=: Object)
Object : class_env(parent=: )
String : class_env(parent=: Object)
WhileIf : class_env(parent=: Object)
method : main paramlist: String[] 
end typing
_______________main___________________
int sum = 1;
    /*
    ------ sum in frame ------
    sum : 1
    */

int i = 1;
    /*
    ------ i in frame ------
    i : 1
    */

sum=sum*i++;
sum=sum*i++;
sum=sum*i++;
sum=sum*i++;
sum=sum*i++;
sum=sum*i++;
sum=sum*i++;
sum=sum*i++;
sum=sum*i++;
sum=sum*i++;
int result = 0;
    /*
    ------ result in frame ------
    result : 0
    */

result=sum;
int r = result;
    /*
    ------ r in frame ------
    r : 3628800
    */

